### PR TITLE
Updated ETH error message for insufficient funds/gas

### PR DESCRIPTION
**Pull request by CLPeterson**

### CODE REVIEW

File: brd-ios/breadwallet/src/ViewControllers/Import/ImportKeyViewController.swift [hunk 1]
```diff
@@ -306,6 +306,8 @@ class ImportKeyViewController: UIViewController, Subscriber {
         switch error {
         case .InsufficientFunds:
             showErrorMessage(L10n.Send.insufficientFunds)
+        case .InsufficientGas:
+            showErrorMessage(L10n.Send.insufficientFunds)
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/ViewControllers/Import/ImportKeyViewController.swift:310]: Are we sure that we should be showing the same message for both InsufficientFunds and InsufficientGas? in that case they can be combined like

case .InsufficientFunds, .InsufficientGas:
    showErrorMessage(L10n.Send.insufficientFunds)

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift [hunk 2]
```diff
@@ -328,13 +328,27 @@ class SendViewController: UIViewController, Subscriber, ModalPresentable {
     }
     
     // TODO: WalletKit is not throwing the right error. Instead, we are always getting "WalletKit.Wallet.FeeEstimationError.ServiceError" error.
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift:330]: This comment can be removed

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift [hunk 3]
```diff
@@ -328,13 +328,27 @@ class SendViewController: UIViewController, Subscriber, ModalPresentable {
...
-    private func handleEstimateFeeError() {
+    private func handleEstimateFeeError(error: Error) {
         sendButton.isEnabled = false
         
         if currency.isERC20Token {
             _ = handleValidationResult(.insufficientGas)
         } else {
-            _ = handleValidationResult(.insufficientFunds)
+            if error as! WalletKit.Wallet.FeeEstimationError == WalletKit.Wallet.FeeEstimationError.InsufficientFunds {
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift:337]: Please, no force unwrapping. Use `as?` instead

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift [hunk 4]
```diff
@@ -328,13 +328,27 @@ class SendViewController: UIViewController, Subscriber, ModalPresentable {
...
+                _ = handleValidationResult(.insufficientFunds)
+            } else if error as! WalletKit.Wallet.FeeEstimationError == WalletKit.Wallet.FeeEstimationError.InsufficientGas {
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift:339]: same here

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift [hunk 5]
```diff
@@ -328,13 +328,27 @@ class SendViewController: UIViewController, Subscriber, ModalPresentable {
...
+                if let amt = amount {
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift:340]: please, no short names. 
let amount = amount is more readable

File: brd-ios/breadwallet/src/ViewControllers/Import/ImportKeyViewController.swift (hunk 1 above)
Comment (CLPeterson) [reply brd-ios/breadwallet/src/ViewControllers/Import/ImportKeyViewController.swift]: A separate general error message was created for insufficient gas and will be displayed for the .InsufficientGas case.

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift (hunk 2 above)
Comment (CLPeterson) [reply brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift]: The comment was removed.

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift (hunk 3 above)
Comment (CLPeterson) [reply brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift]: The forced unwrapping as! was replaced with as?.

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift (hunk 4 above)
Comment (CLPeterson) [reply brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift]: The forced unwrapping as! was replaced with as?.

File: brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift (hunk 5 above)
Comment (CLPeterson) [reply brd-ios/breadwallet/src/ViewControllers/RootModals/SendViewController.swift]: The short name amt was replaced with amount.

**Review by kenanmamedoff — APPROVED** [review]

**Pull request merged**